src/enclave/enclave_oe.c

```diff
@@ -61,40 +178,71 @@ void sgxlkl_ethread_init(void)
     sched_tcb->schedctx = (struct schedctx*)((char*)tls_page + tls_offset);
 
     /* Wait until libc has been initialized */
-    while (__libc_state != libc_initialized)
+    while (sgxlkl_enclave_state.libc_state != libc_initialized)
     {
         a_spin();
     }
 
     /* Initialization completed, now run the scheduler */
     __init_tls();
-    _lthread_sched_init(sgxlkl_enclave->stacksize);
+    _lthread_sched_init(sgxlkl_enclave_state.config->stacksize);
     lthread_run();
 
     return;
 }
 
-int sgxlkl_enclave_init(const sgxlkl_config_t* config_on_host)
+static int _read_eeid_config()
 {
-    SGXLKL_ASSERT(config_on_host);
+    const oe_eeid_t* eeid = (oe_eeid_t*)__oe_get_eeid();
+    const char* config_json = (const char*)eeid->data;
+    sgxlkl_enclave_state.libc_state = libc_not_started;
 
+    sgxlkl_enclave_config_t* cfg = malloc(sizeof(sgxlkl_enclave_config_t));
+    sgxlkl_read_enclave_config(config_json, cfg, true);
+    sgxlkl_enclave_state.config = cfg;
+    return 0;
+}
+
+static int _copy_shared_memory(const sgxlkl_shared_memory_t* shm)
+{
+    // Copy shared memory. Deep copy so the host can't change it?
+    memcpy(
+        &sgxlkl_enclave_state.shared_memory,
+        shm,
+        sizeof(sgxlkl_shared_memory_t));
+
+    // This will be removed once shared memory and config have been
+    // separated fully.
+    sgxlkl_enclave = sgxlkl_enclave_state.config;
+
+    return 0;
+}
+
+int sgxlkl_enclave_init(const sgxlkl_shared_memory_t* shared_memory)
+{
+    SGXLKL_ASSERT(shared_memory);
+
+    memset(&sgxlkl_enclave_state, 0, sizeof(sgxlkl_enclave_state));
+    sgxlkl_enclave_state.libc_state = libc_not_started;
+
+#ifdef DEBUG
+    sgxlkl_verbose = 0;
+#endif
+
+    if (_read_eeid_config())
+        return 1;
+
+    if (_copy_shared_memory(shared_memory))
+        return 1;
```

Comment:
Same comment as above -- what does returning 1 mean?

Reply:
Same as everywhere in the vicinity...

Reply:
Just noticed that this is in `int sgxlkl_enclave_init(...)`. The host assumes this function returns an `oe_result_t`, but it doesn't in the other code paths, so it's not clear what any of this really means.

Reply:
Fixed!